core/trino-main/src/main/java/io/trino/operator/aggregation/partial/SkipAggregationBuilder.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    @Override
+    public Work<?> processPage(Page page)
+    {
+        checkArgument(currentPage == null);
+        currentPage = page;
+        return new CompletedWork<>();
+    }
+
+    @Override
+    public WorkProcessor<Page> buildResult()
+    {
+        if (currentPage == null) {
+            return WorkProcessor.of();
+        }
+
+        Page result = buildOutputPage(currentPage);
+        currentPage = null;
+        return WorkProcessor.of(result);
+    }
+
+    @Override
+    public boolean isFull()
+    {
+        return currentPage != null;
+    }
+
+    @Override
+    public void updateMemory()
+    {
+        if (currentPage != null) {
+            memoryContext.setBytes(currentPage.getSizeInBytes());
+        }
+    }
+
+    @Override
+    public void recordHashCollisions(HashCollisionsCounter hashCollisionsCounter)
+    {
+        // no op
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    @Override
+    public ListenableFuture<Void> startMemoryRevoke()
+    {
+        throw new UnsupportedOperationException("startMemoryRevoke not supported for SkipAggregationBuilder");
+    }
+
+    @Override
+    public void finishMemoryRevoke()
+    {
+        throw new UnsupportedOperationException("finishMemoryRevoke not supported for SkipAggregationBuilder");
+    }
+
+    private Page buildOutputPage(Page page)
+    {
+        populateInitialAccumulatorState(page);
+
+        BlockBuilder[] outputBuilders = serializeAccumulatorState(page.getPositionCount());
+
+        return constructOutputPage(page, outputBuilders);
+    }
+
+    private void populateInitialAccumulatorState(Page page)
+    {
+        GroupByIdBlock groupByIdBlock = getGroupByIdBlock(page.getPositionCount());
+        for (GroupedAggregator groupedAggregator : groupedAggregators) {
+            groupedAggregator.processPage(groupByIdBlock, page);
+        }
+    }
+
+    private GroupByIdBlock getGroupByIdBlock(int positionCount)
+    {
+        return new GroupByIdBlock(
+                positionCount,
+                new LongArrayBlock(positionCount, Optional.empty(), consecutive(positionCount)));
+    }
+
+    private BlockBuilder[] serializeAccumulatorState(int positionCount)
+    {
+        BlockBuilder[] outputBuilders = new BlockBuilder[groupedAggregators.size()];
+        for (int i = 0; i < outputBuilders.length; i++) {
+            outputBuilders[i] = groupedAggregators.get(i).getType().createBlockBuilder(null, positionCount);
+        }
+
+        for (int position = 0; position < positionCount; position++) {
+            for (int i = 0; i < groupedAggregators.size(); i++) {
+                GroupedAggregator groupedAggregator = groupedAggregators.get(i);
+                BlockBuilder output = outputBuilders[i];
+                groupedAggregator.evaluate(position, output);
+            }
+        }
+        return outputBuilders;
+    }
+
+    private Page constructOutputPage(Page page, BlockBuilder[] outputBuilders)
```

Comment:
technically, your output page might become quite large since we add more columns. However, this probably isn't a big issue.
In `io.trino.operator.project.PageProcessor` we avoid this by keeping page under 4MB